packages/pds/src/sequencer/outbox.ts

```diff
@@ -104,19 +104,44 @@ export class Outbox {
 
   // yields only historical events
   async *getBackfill(backfillCursor: number, backfillTime?: string) {
+    const PAGE_SIZE = 200
+    const [firstEvt] = await this.sequencer.requestSeqRange({
+      earliestTime: backfillTime,
+      earliestSeq: this.lastSeen > -1 ? this.lastSeen : backfillCursor,
+      limit: 1,
+    })
+    if (firstEvt) {
+      yield firstEvt
+    } else {
+      return
+    }
+
+    let nextPage = this.sequencer.requestSeqRange({
+      earliestSeq: this.lastSeen,
+      limit: PAGE_SIZE,
+    })
+    let queued = this.sequencer.requestSeqRange({
+      earliestSeq: this.lastSeen + PAGE_SIZE,
+      limit: PAGE_SIZE,
+    })
+
     while (true) {
-      const evts = await this.sequencer.requestSeqRange({
-        earliestTime: backfillTime,
-        earliestSeq: this.lastSeen > -1 ? this.lastSeen : backfillCursor,
-        limit: 100,
-      })
+      const evts = await nextPage
       for (const evt of evts) {
-        yield evt
+        if (evt.seq > this.lastSeen) {
+          yield evt
+        }
       }
-      // if we're within 50 of the sequencer, we call it good & switch to cutover
+      // if we're within half a pagesize of the sequencer, we call it good & switch to cutover
       const seqCursor = this.sequencer.lastSeen ?? -1
-      if (seqCursor - this.lastSeen < 100) break
+      if (seqCursor - this.lastSeen < PAGE_SIZE / 2) break
       if (evts.length < 1) break
+
+      nextPage = queued
+      queued = this.sequencer.requestSeqRange({
+        earliestSeq: this.lastSeen + PAGE_SIZE,
+        limit: PAGE_SIZE,
+      })
```

Comment:
I am currently a little fuzzy on exactly how this will pan out, i.e. what kinds of cases could arise if there's a bug.  I think the thing that makes it trickier for me is that `this.lastSeen` is core to the pagination happening here but it's set elsewhere.

For example, I see the defensive `evt.seq > this.lastSeen` check to ensure we don't put the same event out on the stream twice (e.g. when there's a gap on one of the pages and both `nextPage` and `queued` contain some of the same events), and correctness there depends on `lastSeen` being incremented elsewhere.  Another case that might happen if `lastSeen` isn't incremented would be getting stuck in a loop for the same page.

Don't think we need to block on this, but I bet if we chat it through I'll have a better sense for it.